central/node/datastore/store/postgres/cve_store.go

```diff
@@ -0,0 +1,228 @@
+		}
+		msg := &storage.NodeCVE{}
+		if err := msg.UnmarshalVTUnsafe(data); err != nil {
+			return nil, errors.Wrap(err, "unmarshaling CVE data")
+		}
+		dbCVEs[msg.GetId()] = msg
+		idToCVEMap[msg.GetId()] = msg
+	}
+
+	// Cache the newly fetched CVEs
+	if len(dbCVEs) > 0 {
+		s.cache.SetMany(dbCVEs)
+	}
+
+	if err := rows.Err(); err != nil {
+		return nil, errors.Wrap(err, "iterating over CVE rows")
+	}
+
+	return idToCVEMap, nil
+}
+
+func (s *nodeCVEStoreImpl) CopyFromNodeCves(ctx context.Context, tx *postgres.Tx, objs ...*storage.NodeCVE) error {
+	copyCols := []string{
+		"id",
+		"cvebaseinfo_cve",
+		"cvebaseinfo_publishedon",
+		"cvebaseinfo_createdat",
+		"operatingsystem",
+		"cvss",
+		"severity",
+		"impactscore",
+		"snoozed",
+		"snoozeexpiry",
+		"orphaned",
+		"orphanedtime",
+		"serialized",
+	}
+
+	// Process CVEs in batches using slices.Chunk
+	for batch := range slices.Chunk(objs, batchSize) {
+		// Prepare data for this batch
+		inputRows := make([][]interface{}, 0, len(batch))
+		deletes := make([]string, 0, len(batch))
+
+		for _, obj := range batch {
+			serialized, marshalErr := obj.MarshalVT()
+			if marshalErr != nil {
+				return errors.Wrapf(marshalErr, "marshaling CVE %s", obj.GetId())
+			}
+
+			inputRows = append(inputRows, []interface{}{
+				obj.GetId(),
+				obj.GetCveBaseInfo().GetCve(),
+				protocompat.NilOrTime(obj.GetCveBaseInfo().GetPublishedOn()),
+				protocompat.NilOrTime(obj.GetCveBaseInfo().GetCreatedAt()),
+				obj.GetOperatingSystem(),
+				obj.GetCvss(),
+				obj.GetSeverity(),
+				obj.GetImpactScore(),
+				obj.GetSnoozed(),
+				protocompat.NilOrTime(obj.GetSnoozeExpiry()),
+				obj.GetOrphaned(),
+				protocompat.NilOrTime(obj.GetOrphanedTime()),
+				serialized,
+			})
+
+			deletes = append(deletes, obj.GetId())
+		}
+
+		// Copy does not upsert so have to delete first.
+		_, err := tx.Exec(ctx, "DELETE FROM "+nodeCVEsTable+" WHERE id = ANY($1::text[])", deletes)
+		if err != nil {
+			return errors.Wrap(err, "deleting CVEs before copy")
+		}
+
+		// Invalidate cache for deleted CVEs
+		if len(deletes) > 0 {
+			s.cache.DeleteMany(deletes)
+		}
+
+		// Insert the batch
+		_, err = tx.CopyFrom(ctx, pgx.Identifier{nodeCVEsTable}, copyCols, pgx.CopyFromRows(inputRows))
+		if err != nil {
+			return errors.Wrap(err, "copying CVE batch to database")
+		}
+	}
+
+	// Update cache with all successfully inserted/updated CVEs
+	cveMap := make(map[string]*storage.NodeCVE)
+	for _, obj := range objs {
+		cveMap[obj.GetId()] = obj
+	}
+	s.cache.SetMany(cveMap)
+
+	return nil
+}
+
+func (s *nodeCVEStoreImpl) RemoveOrphanedNodeCVEs(ctx context.Context, tx *postgres.Tx) error {
+	// Delete orphaned CVEs and return their IDs for cache invalidation
+	rows, err := tx.Query(ctx, "DELETE FROM "+nodeCVEsTable+" WHERE not exists (select "+componentCVEEdgesTable+".nodecveid from "+componentCVEEdgesTable+" where "+nodeCVEsTable+".id = "+componentCVEEdgesTable+".nodecveid) RETURNING id")
```

Comment:
nitpick: let's keep it consistent. (didn't check other SQL statements)
 
```suggestion
	rows, err := tx.Query(ctx, "DELETE FROM "+nodeCVEsTable+" WHERE NOT EXISTS (SELECT "+componentCVEEdgesTable+".nodecveid FROM "+componentCVEEdgesTable+" WHERE "+nodeCVEsTable+".id = "+componentCVEEdgesTable+".nodecveid) RETURNING id")
```